src/aomusicplayer.cpp

```diff
@@ -204,13 +204,19 @@ void AOMusicPlayer::set_looping(bool toggle, int channel)
     }
     if (loop_start[channel] > 0) {
       if (loop_end[channel] == 0)
-        loop_end[channel] =
-            BASS_ChannelGetLength(m_stream_list[channel], BASS_POS_BYTE);
-      if (loop_end[channel] >
-          0) // Don't loop zero length songs even if we're asked to
+      {
+        // We don't have length, loop when track reaches end
+        loop_sync[channel] = BASS_ChannelSetSync(
+            m_stream_list[channel], BASS_SYNC_END | BASS_SYNC_MIXTIME,
+            loop_end[channel], loopProc, &loop_start[channel]);
```

Comment:
```suggestion
        loop_sync[channel] = BASS_ChannelSetSync(
            m_stream_list[channel], BASS_SYNC_END | BASS_SYNC_MIXTIME,
            0, loopProc, &loop_start[channel]);
```

I think there's a way to merge both of these branches by using [`BASS_ChannelSetPosition`](http://www.un4seen.com/doc/#bass/BASS_ChannelSetPosition.html) where `pos` is `BASS_POS_END`. Then it's possible to use ChannelSetSync with `BASS_SYNC_END` per the docs:

> BASS_SYNC_END - Sync when a channel reaches the end, including when looping
> ...
> data : 0 = the normal end position, 1 = a backward jump in a MOD music, **2 = the BASS_POS_END position**, 3 = the end of a tail (BASS_ATTRIB_TAIL).

Then you no longer have to worry about invalid loop points.